sdk/core/core-client/src/caeCallbacks.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { GetTokenOptions } from "@azure/core-auth";
+import { AuthorizeRequestOnChallengeOptions, ChallengeCallbacks } from "@azure/core-rest-pipeline";
+import { decodeString } from "./base64";
+
+/**
+ * Converts: `Bearer a="b", c="d", Bearer d="e", f="g"`.
+ * Into: `[ { a: 'b', c: 'd' }, { d: 'e', f: 'g"' } ]`.
+ *
+ * Important:
+ * Do not use this in production, as values might contain the strings we use to split things up.
+ *
+ * @internal
+ */
+function parseCAEChallenge(challenges: string): any[] {
+  return challenges
+    .split("Bearer ")
+    .filter((x) => x)
+    .map((challenge) =>
+      `${challenge.trim()}, `
+        .split('", ')
+        .filter((x) => x)
+        .map((keyValue) => (([key, value]) => ({ [key]: value }))(keyValue.trim().split('="')))
+        .reduce((a, b) => ({ ...a, ...b }), {})
+    );
+}
+
+/**
+ * CAE Challenge structure
+ */
+export interface CAEChallenge {
+  scope: string;
+  claims: string;
+}
+
+/**
+ * Generates callbacks to use on the `bearerTokenAuthenticationPolicy` to support CAE challenges:
+ * [Continuous Access Evaluation](https://docs.microsoft.com/azure/active-directory/conditional-access/concept-continuous-access-evaluation).
+ */
+export function createCAECallbacks(): ChallengeCallbacks {
+  let cachedChallenge: string | undefined;
+
+  return {
+    async authorizeRequestOnChallenge(
```

Comment:
Another option would be exporting this function (maybe with a better name), then consuming it like `bearerTokenAuthenticationPolicy(credential, scopes, challengeCallbacks: { authorizeRequestOnChallenge: authorizeRequestOnClaimChallenge })`

Not sure which approach is better.

Reply:
well, the issue is that we can’t just export the single function, because we need to keep state. I need to update this PR to better reflect that. I’ll do that later today.

Reply:
oh my bad, we actually can export just this function. I was mixing the test scenario for challenges, that meant to showcase stateful challenges, with the CAE specific one. I’ll update this PR.